[Platform] Parse concatenated QR code is failing in Darwin Framework Tool

### Reproduction steps

Please follow the steps of TC-DD1.11 where we run all cluster app on two Test Harnesses and give the respective QR codes obtained from the two of them to the DUT to parse. 

Chip tool was able to parse the concatenated QR code and give individual passcodes but Darwin was not able to do that. Below is the log for both,


shyam@Shyams-MacBook-Pro debug % ./chip-tool payload parse-setup-payload "MT:-24J0CEK01KA0648G00*-24J0KQS020O0648G00"
[1772822222.442] [22900:321897:main] [DL] KVS will be written to: file:///var/folders/rn/r3b6zlms0vvbc9zwhqhjrzbc0000gn/T//chip_tool_kvs
[1772822222.455] [22900:321897:main] [SPL] Parsing base38Representation: MT:-24J0CEK01KA0648G00*-24J0KQS020O0648G00
[1772822222.455] [22900:321897:main] [SPL] Version:             0
[1772822222.455] [22900:321897:main] [SPL] VendorID:            65521
[1772822222.455] [22900:321897:main] [SPL] ProductID:           32769
[1772822222.455] [22900:321897:main] [SPL] Custom flow:         0    (STANDARD)
[1772822222.455] [22900:321897:main] [SPL] Discovery Bitmask:   0x02 (BLE)
[1772822222.455] [22900:321897:main] [SPL] Long discriminator:  3841   (0xf01)
[1772822222.455] [22900:321897:main] [SPL] Passcode:            20202021
[1772822222.455] [22900:321897:main] [SPL] ----------
[1772822222.455] [22900:321897:main] [SPL] Version:             0
[1772822222.455] [22900:321897:main] [SPL] VendorID:            65521
[1772822222.455] [22900:321897:main] [SPL] ProductID:           32769
[1772822222.455] [22900:321897:main] [SPL] Custom flow:         0    (STANDARD)
[1772822222.455] [22900:321897:main] [SPL] Discovery Bitmask:   0x02 (BLE)
[1772822222.455] [22900:321897:main] [SPL] Long discriminator:  3842   (0xf02)
[1772822222.455] [22900:321897:main] [SPL] Passcode:            20202022
shyam@Shyams-MacBook-Pro debug % cd ..
shyam@Shyams-MacBook-Pro out % cd ..
shyam@Shyams-MacBook-Pro chip-tool % cd ..
shyam@Shyams-MacBook-Pro examples % cd darwin-framework-tool/out/debug 
shyam@Shyams-MacBook-Pro debug % ./darwin-framework-tool payload parse-setup-payload "MT:-24J0CEK01KA0648G00*-24J0KQS020O0648G00"
2026-03-06 10:37:25.589 darwin-framework-tool[22901:322373] Version:       0
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] VendorID:      65521
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] ProductID:     32769
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] Custom flow:   0    (STANDARD)
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] Capabilities:  0x02 (BLE)
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] Discriminator: 3841
2026-03-06 10:37:25.590 darwin-framework-tool[22901:322373] Passcode:      20202021
shyam@Shyams-MacBook-Pro debug % ./darwin-framework-tool pairing                                                                 
2026-03-06 10:38:47.115 [22918:4103:com.chip.main.dft] [TOO]: Missing command name
Usage:
  ./darwin-framework-tool pairing command_name [param1 param2 ...]

Commands for commissioning devices.

  +-------------------------------------------------------------------------------------+
  | Commands:                                                                           |
  +-------------------------------------------------------------------------------------+
  | * code                                                                              |
  | * code-paseonly                                                                     |
  | * code-wifi                                                                         |
  | * code-thread                                                                       |
  | * ble-wifi                                                                          |
  | * ble-thread                                                                        |
  | * by-index                                                                          |
  | * by-index-paseonly                                                                 |
  | * unpair                                                                            |
  | * open-commissioning-window                                                         |
  | * pre-warm-commissioning                                                            |
  | * get-commissioner-node-id                                                          |
  +-------------------------------------------------------------------------------------+
2026-03-06 10:38:47.120 [22918:4103:com.chip.main.dft] [TOO]: Run command failure: examples/chip-tool/commands/common/Commands.cpp:248: Error 0x0000002F
shyam@Shyams-MacBook-Pro debug % 


Additionally, on network pairing is not available in Darwin Framework Tool for the second step of this testcase where we use the obtained passcodes from the first step to do pairing. 

### Platform

darwin

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

--- a/examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm
+++ b/examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm
@@ -69,6 +69,20 @@ CHIP_ERROR SetupPayloadParseCommand::Run()
 
 CHIP_ERROR SetupPayloadParseCommand::Print(MTRSetupPayload * payload)
 {
+    // If the top-level payload is concatenated, we recurse with the subpayloads
+    //  which will have `isConcatenated` as `false`. 
+    if (payload.isConcatenated) {
+        bool firstPayload = true;
+        for (MTRSetupPayload * subPayload in payload.subPayloads) {
+            if (!firstPayload) {
+                NSLog(@"----------");
+            }
+            ReturnErrorOnFailure(Print(subPayload));
+            firstPayload = false;
+        }
+        return CHIP_NO_ERROR;
+    }
+
     NSLog(@"Version:       %@", payload.version);
     NSLog(@"VendorID:      %@", payload.vendorID);
     NSLog(@"ProductID:     %@", payload.productID);

--- a/examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm
+++ b/examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm
@@ -70,7 +70,7 @@ CHIP_ERROR SetupPayloadParseCommand::Run()
 CHIP_ERROR SetupPayloadParseCommand::Print(MTRSetupPayload * payload)
 {
     // If the top-level payload is concatenated, we recurse with the subpayloads
-    //  which will have `isConcatenated` as `false`. 
+    //  which will have `isConcatenated` as `false`.
     if (payload.isConcatenated) {
         bool firstPayload = true;
         for (MTRSetupPayload * subPayload in payload.subPayloads) {

